### PR TITLE
feat(Components/TMC-16845): badge visual update

### PR DESCRIPTION
**Pull request by smouillour**
**What is the problem this PR is trying to solve?**
https://jira.talendforge.org/browse/TMC-16845

Previous visual
image

New visual for badges
image

New visual for badges with option white (to manage case of grey background)
image

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/Badge/Badge.component.js [hunk 1]
```diff
@@ -19,32 +21,87 @@ function Badge({
...
 }) {
+	const displayClass = display === 'small' ? 'tc-badge-display-small' : 'tc-badge-display-large';
+
 	const containerClasses = classNames(
 		'tc-badge',
 		theme['tc-badge'],
 		selected && ['tc-badge-selected', theme['tc-badge-selected']],
 		disabled && ['tc-badge-disabled', theme['tc-badge-disabled']],
 		!onDelete && ['tc-badge-readonly', theme['tc-badge-readonly']],
+		[displayClass, theme[displayClass]],
```
Comment (romainseb) [inline packages/components/src/Badge/Badge.component.js]: i suggest you to use the css util that i've set on skeleton component (https://github.com/Talend/ui/blob/master/packages/components/src/Skeleton/Skeleton.component.js#L10)
(https://github.com/Talend/ui/pull/2109)

That means you could use it this way :
```javascript
// top of the file
import badgeCssModule from './Badge.scss';
import { getTheme } from '../theme';

const theme = getTheme(badgeCssModule);

// here
const containerClasses = theme(
  'tc-badge',
  displayClass,
  className,
  {
    'tc-badge-selected': selected,
    'tc-badge-disabled': disabled,
    'tc-badge-readonly': !onDelete,
    'tc-badge-aslink': aslink,
    'tc-badge-edit': edit,
  }
)
```

It allow to not repeat the class twice & using the classname style api with the object to apply the classes or not

File: packages/components/src/Badge/Badge.component.js [hunk 2]
```diff
@@ -19,32 +21,87 @@ function Badge({
...
+		aslink && ['tc-badge-aslink', theme['tc-badge-aslink']],
+		edit && ['tc-badge-edit', theme['tc-badge-edit']],
 		className,
 	);
 	const badgeClasses = classNames('tc-badge-button', theme['tc-badge-button']);
-	const labelClasses = classNames('tc-badge-label', theme['tc-badge-label']);
+	const badgeWhiteClasses = classNames(
```
Comment (romainseb) [inline packages/components/src/Badge/Badge.component.js]: Could be only badgeClasses if you use the classname api

```javascript
const badgeClasses = theme(
  'tc-badge-button',
  {
    'tc-badge-white': white,
  }
)
```

File: packages/components/src/Badge/Badge.component.js (hunk 1 above)
Comment (romainseb) [reply packages/components/src/Badge/Badge.component.js]: Of course, it applies for each classename const added in the file :)

File: packages/components/src/Badge/Badge.scss [hunk 3]
```diff
@@ -1,121 +1,244 @@
...
-		left: -$padding-large;
-
-		svg {
-			display: block;
-			margin: 0;
-			height: $tc-badge-delete-icon-size;
-			width: $tc-badge-delete-icon-size;
+	&-disabled {
+		.tc-badge-button {
+			opacity: 0.62;
```
Comment (romainseb) [inline packages/components/src/Badge/Badge.scss]: 0.62 opacity ? seems very specific 😅
Is it wanted ? If yes, maybe we could put this in a variable top of the file ? 😇

File: packages/components/src/Badge/Badge.component.js [hunk 4]
```diff
@@ -19,32 +21,87 @@ function Badge({
 	t,
 	style,
 	className,
+	display = 'large',
```
[the remark this answers is not in the document]
Comment (smouillour) [reply packages/components/src/Badge/Badge.component.js]: 👍

File: packages/components/src/Badge/Badge.component.js [hunk 5]
```diff
@@ -19,32 +21,87 @@ function Badge({
...
+	aslink,
+	edit,
+	icon,
+	white,
 }) {
+	const displayClass = display === 'small' ? 'tc-badge-display-small' : 'tc-badge-display-large';
```
[the remark this answers is not in the document]
Comment (smouillour) [reply packages/components/src/Badge/Badge.component.js]: :+1:

File: packages/components/src/Badge/Badge.component.js (hunk 1 above)
Comment (smouillour) [reply packages/components/src/Badge/Badge.component.js]: 👍

File: packages/components/src/Badge/Badge.component.js (hunk 2 above)
Comment (smouillour) [reply packages/components/src/Badge/Badge.component.js]: 👍

File: packages/components/src/Badge/Badge.component.js [hunk 6]
```diff
@@ -19,28 +28,67 @@ function Badge({
...
+		category && <span className={separatorClasses} />,
+		<div className={labelClasses}>
+			<TooltipTrigger label={label} tooltipPlacement="top">
+				<span key="label" className={labelTextClasses}>
+					{label}
+				</span>
+			</TooltipTrigger>
+			{icon && <Icon name={icon} className={labelIconClasses} />}
+		</div>,
+		icon && <span className={[separatorIconClasses]} />,
```
Comment (romainseb) [inline packages/components/src/Badge/Badge.component.js]:
```suggestion
		icon && <span className={theme('tc-badge-separator', 'tc-badge-separator-icon')} />,
```

File: packages/components/src/Badge/Badge.component.js [hunk 7]
```diff
@@ -19,28 +28,67 @@ function Badge({
...
-		</span>,
+		category && (
+			<TooltipTrigger label={category} tooltipPlacement="top">
+				<span key="category" aria-label={category} className={categoryClasses}>
+					{category}
+				</span>
+			</TooltipTrigger>
+		),
+		category && <span className={separatorClasses} />,
+		<div className={labelClasses}>
```
Comment (romainseb) [inline packages/components/src/Badge/Badge.component.js]:
```suggestion
		<div className={theme('tc-badge-label')}>
```

File: packages/components/src/Badge/Badge.component.js [hunk 8]
```diff
@@ -19,28 +28,67 @@ function Badge({
...
+			<TooltipTrigger label={label} tooltipPlacement="top">
+				<span key="label" className={labelTextClasses}>
+					{label}
+				</span>
+			</TooltipTrigger>
+			{icon && <Icon name={icon} className={labelIconClasses} />}
```
Comment (romainseb) [inline packages/components/src/Badge/Badge.component.js]:
```suggestion
			{icon && <Icon name={icon} className={theme('tc-badge-label-icon')} />}
```

**Pull request merged**